osu.Game/Overlays/Chat/ChatTextBar.cs

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Online.Chat;
+using osuTK;
+
+namespace osu.Game.Overlays.Chat
+{
+    public class ChatTextBar : Container
+    {
+        public readonly BindableBool ShowSearch = new BindableBool();
+
+        public ChatTextBox TextBox => chatTextBox;
```

Comment:
I'm unsure if exposing the child text box like this is a good approach, but was unsure of how else to have the textbox events being available to bind to from the parent component while keeping everything self-contained :/

Reply:
What does this mean? Not sure I follow. Right now this public text box seems to only be exposed for the test which is definitely not a good enough reason for it to be public.

If you wish to re-expose events for something else, then the way to do that generally would be to re-expose the event and internally proxy:

```csharp
public class ChatTextBar
{
    public event TextBox.OnCommitHandler OnCommit;

    [BackgroundDependencyLoader]
    private void load()
    {
        // ...
        textBox.OnCommit += (sender, newText) => OnCommit?.Invoke(sender, newText);
    }
    // ...
}
```

Compare: [`LabelledTextBox`](https://github.com/ppy/osu/blob/0020102810c8c3c2c4b8a319781c0f2ceb74dd48/osu.Game/Graphics/UserInterfaceV2/LabelledTextBox.cs#L67)

Reply:
Apologies for not being clear earlier. I exposed the textbox publicly because when this component is added to the overlay it would need to be able to access the events and properties of the textbox for message handling and such as the logic for message sending would be at the overlay component level.

As such you wound need to manually forward the texbox `OnCommit`, `Current` bindable (for interactive search), and focus handling functions. I figured it would have been simpler to just expose the textbox rather than re-exposing each of these properties. However if re-exposing is the preferred way I will do that.

Reply:
I see two issues here:

* For the uses that you are looking for, directly binding to text box events or even proxying seems like possibly approaching things at too low a level. Within the current paramters I don't see a reason why `ChatTextBar` couldn't expose higher-level events like `ChatMessageCommitted` or `ChannelSearchTermsChanged` or similar, because that's what the chat overlay _actually_ cares about, rather than raw commit/current change events.
* But the above proposal - having those two events in one class - is making me think that possibly jamming them into one component at this level was a misstep, and they should be separated, and instead managed/toggled at the chat overlay level rather than via this `ChatTextBar` semi-proxy.

Reply:
* Exposing higher level events actually makes way more sense :+1: 
* I did consider having two textboxes managed at the overlay level, however I felt that it would be a much larger and complex PR to review since there are other components to consider when making changes at the overlay level. I would rather PR smaller self-container components like this to assemble later.